flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -575,12 +606,13 @@ llvm::Optional<mlir::FuncOp> searchFunctionInLibrary(
 /// Search runtime for the best runtime function given an intrinsic name
 /// and interface. The interface may not be a perfect match in which case
 /// the caller is responsible to insert argument and return value conversions.
-static llvm::Optional<mlir::FuncOp>
-getRuntimeFunction(mlir::Location loc, Fortran::lower::FirOpBuilder &builder,
-                   llvm::StringRef name, mlir::FunctionType funcType) {
+static mlir::FuncOp getRuntimeFunction(mlir::Location loc,
+                                       Fortran::lower::FirOpBuilder &builder,
+                                       llvm::StringRef name,
+                                       mlir::FunctionType funcType) {
   const RuntimeFunction *bestNearMatch = nullptr;
   FunctionDistance bestMatchDistance{};
-  llvm::Optional<mlir::FuncOp> match;
+  mlir::FuncOp match;
```

Comment:
Is this because it cannot fail or because we're exploiting a nullptr value? The doxygen comment doesn't make it clear either.

Reply:
because we're exploiting a nullptr value. I added comments.